src/plugin.js

```diff
@@ -5,48 +5,80 @@ import PrismicDOM from 'prismic-dom'
 const moduleOptions = <%= serialize(options) %>
 
 export default async ({ app, req, route, res, query, redirect }, inject) => {
-  let options = {}
+  let options = {};
 
   // Pass through server requests, primarily for preview
   if (process.server) {
     options.req = req
   }
 
-  let api = await Prismic.api(moduleOptions.endpoint, options)
+  let api = await Prismic.api(moduleOptions.endpoint, options);
 
   let prismic = new Vue({
     computed: {
       api() {
         return api
       },
-
+      predicates() {
+        return Prismic.Predicates
+      },
       dom() {
         return PrismicDOM
       }
     },
     methods: {
+      asHtml(richText) {
+        if (richText) {
+          return PrismicDOM.RichText.asHtml(
+            richText,
+            moduleOptions.linkResolver
+          )
+        }
+      },
+      asText(richText) {
+        if (richText) {
+          return PrismicDOM.RichText.asText(richText)
+        }
+      },
+      asLink(link) {
+        if (link) {
+          return PrismicDOM.Link.url(link, moduleOptions.linkResolver)
+        }
+      },
+      asDate(date) {
+        if (date) {
+          return PrismicDOM.Date(date)
+        }
+      },
       linkResolver: moduleOptions.linkResolver
     }
-  })
+  });
 
-  inject('prismic', prismic)
+  inject('prismic', prismic);
 
   // Preview support
-  if (route.path == "/preview") {
+  if (route.path === '/preview') {
     const { token } = query;
 
     if (token) {
       let url = await api.previewSession(token, moduleOptions.linkResolver, '/');
 
+      let c_token = `${Prismic.previewCookie}=${token};`;
+      let c_max_age = `max-age=${30 * 60 * 1000};`;
+      let c_path = 'path=/';
+      let cookie = `${c_token} ${c_max_age} ${c_path}`;
```

Comment:
Not a huge fan of splitting the cookie across multiple variables. I'd prefer to just have this in a one-liner, makes it cleaner and the variable names less ambiguous.

```suggestion
      let cookie = `${Prismic.previewCookie}=${token}; max-age=${30 * 60 * 1000}; path=/`;
```

Or even something like this to make the contents of the cookie a bit easier to read;
```suggestion
      let cookie = [
          `${Prismic.previewCookie}=${token}`,
          `max-age=${30 * 60 * 1000}`,
          'path=/'
      ].join("; ");
```

Reply:
@jamespegg @carlos-avila what can I do to help get this pull request merged in?  Would love to have these prismicDom helpers available as methods, I think it would really improve the dev workflow

Reply:
I got busy during the holiday but I'll push an update today with the changes James suggested.